lib/facter/pulp_consumer_id.rb

```diff
@@ -0,0 +1,14 @@
+Facter.add(:pulp_consumer_id) do
+  confine kernel: 'Linux'
+  setcode do
+    pulp_consumer_id = nil
+    status = Facter::Util::Resolution.exec('pulp-consumer status')
+    unless status.nil?
+      # Strip color from command output
+      status.gsub!(/\e\[([;\d]+)?m/, '')
+      captures = /^This consumer is registered to the server\s\[(.*)\]\swith\sthe\sID\s\[(.*)\]\.$/.match(status)
```

Comment:
Do you want to do something with match[0] - like register an additional fact for the pulp server?